File: python_workspace/cardImageGenerator/generate_prompts.js
const fs = require('fs');
const path = require('path');

// Read TSV file
const content = fs.readFileSync('fighter.tsv', 'utf-8');
const lines = content.split('\n');

// Get header
const header = lines[0].split('\t');
const titleIdx = header.findIndex(col => col.includes('title'));

console.log('Title column index:', titleIdx);

// Extract unique titles with their first occurrence row number
const titleMap = new Map();
for (let i = 1; i < lines.length; i++) {
    const parts = lines[i].split('\t');
    if (parts.length > titleIdx && parts[titleIdx]) {
        const title = parts[titleIdx];
        if (!titleMap.has(title)) {
            titleMap.set(title, i);
        }
    }
}

console.log(`Found ${titleMap.size} unique titles`);

// Create imagePrompt directory
const outputDir = 'imagePrompt';
if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir);
}

// Generate JSON files for each title
let count = 0;
for (const [title, rowNum] of [...titleMap.entries()].sort((a, b) => a[1] - b[1])) {
    // Use row number as filename to preserve order
    const filename = path.join(outputDir, `${String(count + 1).padStart(3, '0')}_${title}.json`);
    
    const promptData = {
        title: title,
        prompt: `${title}, martial arts weapon, traditional Chinese martial arts style, ink wash painting style, sumi-e, monochrome ink painting, flowing brush strokes, dynamic pose, ancient weapon aesthetic, minimalist composition, traditional Asian art style`
    };
    
    fs.writeFileSync(filename, JSON.stringify(promptData, null, 2), 'utf-8');
    count++;
}

console.log(`Generated ${count} prompt files`);

File: python_workspace/cardImageGenerator/generate_prompts.py
import json
import os

# Read TSV file manually
titles = set()
with open('fighter.tsv', 'r', encoding='utf-8') as f:
    lines = f.readlines()
    
# First line is header
header = lines[0].strip().split('\t')
print('Header:', header)

# Find title column index
title_idx = None
for i, col in enumerate(header):
    if 'title' in col.lower():
        title_idx = i
        break

print('Title column index:', title_idx)

# Read data lines
for line in lines[1:]:
    parts = line.strip().split('\t')
    if len(parts) > title_idx and parts[title_idx]:
        titles.add(parts[title_idx])

print(f'Found {len(titles)} unique titles')

# Create imagePrompt directory if not exists
os.makedirs('imagePrompt', exist_ok=True)

# Generate prompts for each unique title
for title in sorted(titles):
    # Create safe filename
    safe_name = ''.join(c if c.isalnum() or c in '-_' else '_' for c in title)
    filename = f'imagePrompt/{safe_name}.json'
    
    prompt_data = {
        'title': title,
        'prompt': f'{title}, martial arts weapon, traditional Chinese martial arts style, ink wash painting style, sumi-e, monochrome ink painting, flowing brush strokes, dynamic pose, ancient weapon aesthetic, minimalist composition, traditional Asian art style'
    }
    
    with open(filename, 'w', encoding='utf-8') as f:
        json.dump(prompt_data, f, ensure_ascii=False, indent=2)

print(f'Generated {len(titles)} unique title prompts')